src/js/components/receiving/PartialReceivingPage.jsx

```diff
@@ -475,9 +475,30 @@ class PartialReceivingPage extends Component {
     const url = `/openboxes/api/partialReceiving/${this.props.match.params.shipmentId}?stepNumber=1`;
 
     return apiClient.get(url)
-      .then((response) => {
+      .then((resp) => {
+        const response = parseResponse(resp.data.data);
         this.setState({ values: {} }, () => {
-          this.setState({ values: parseResponse(response.data.data) });
+          this.setState({
+            values: {
+              ...response,
+              containers: _.map(response.containers, container => ({
+                ...container,
+                shipmentItems: _.map(container.shipmentItems, shipmentItem => ({
+                  ...shipmentItem,
+                  binLocation: {
+                    ...shipmentItem.binLocation,
+                    value: shipmentItem.binLocation ? shipmentItem.binLocation.id : null,
```

Comment:
do we have to add value and label everywhere, for the select, I think it might be configurable, so you can specify the label field name and value field name in the select properties